### ISSUE TITLE
Android build failing for `0.18.0`

### ISSUE THREAD

**Issue by fdietze** [luanpotter/audioplayers#810]
Since version `0.18.0`, the android build is failing:

```
$ flutter run
Launching lib/main.dart on Android SDK built for x86 in debug mode...
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (18, 32): Unresolved reference: mutableMapOf
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (18, 45): Cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (25, 9): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (26, 9): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (27, 9): Cannot access built-in declaration 'kotlin.Boolean'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (27, 9): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (28, 17): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (34, 13): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (36, 20): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (37, 22): Cannot access built-in declaration 'kotlin.Unit'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (37, 51): Cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (42, 29): Cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (42, 38): Cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library
e: /home/felix/development/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.18.0/android/src/main/kotlin/xyz/luan/audioplayers/AudioplayersPlugin.kt: (42, 58): Cannot access built-in declaration 'kotlin.String'. Ensure that you have a dependency on the Kotlin standard library

[...]
```

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.2, on Linux, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] VS Code (version 1.52.1)
[✓] Connected device (2 available)

• No issues found!
```

**Comment by luanpotter** [comment]
Could you please send the contents of both your gradle files? I believe you might be overriding some kotlin configuration (which is weird because each library should be independent).

Another question, if you run `flutter clean && flutter pub get` do you still get the issue? Same error?

Another question, can you try running the example app on your setup?

**Comment by luanpotter** [comment]
Also, have you tried this? https://github.com/luanpotter/audioplayers/issues/790

**Comment by pankajadhyapak** [comment]
faced same issue, fixed by updating `ext.kotlin_version` to `1.4.31` in `android/build.gradle`.

**Comment by jhonatanluiz** [comment]
> faced same issue, fixed by updating `ext.kotlin_version` to `1.4.31` in `android/build.gradle`.

it worked for me! thanks!

**Comment by dvorapa** [comment]
The problem with this solution is, that `flutter create` (on stable channel, currently Flutter 2.0.3) still creates new projects with `ext.kotlin_version = '1.3.50'`, therefore every new app is still going to face this issue.

**Comment by fdietze** [comment]
Just got back to this issue. Thanks everyone for the debugging, I'll test the PR now :)

**Comment by luanpotter** [comment]
Merged a fix, will do some testing and release 18.1

**Comment by itssidhere** [comment]
I am not using kotlin in my project and still getting the same error even after upgrading to 0.18.1 . Any workarounds?

**Comment by luanpotter** [comment]
@itssidhere have you tried all the fixes mentioned above? I just released 0.18.2 can you please try that version?

**Comment by erickzanardo** [comment]
#811 should have fixed this. If the error persists, let us know and we can reopen this.
